src/rpc/blockchain.cpp

```diff
@@ -1515,8 +1514,18 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
         }
     }
 
-    if (blockcount < 1 || blockcount >= pindex->nHeight) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid block count: should be between 1 and the block's height");
+    if (pindex->nHeight < 1) {
```

Comment:
@promag Thanks for a good suggestion.
@promag @jnewbery Is it better to reverse the conditions?
```c++
if (request.params[0].isNull()) {
    blockcount = std::max(0, std::min(blockcount, pindex->nHeight - 1));
} else {
    ...
}
```